akka-actor/src/main/scala/akka/actor/LightArrayRevolverScheduler.scala

```diff
@@ -86,7 +86,7 @@ class LightArrayRevolverScheduler(config: Config, log: LoggingAdapter, threadFac
     val sleepMs = if (Helpers.isWindows) (nanos + 4999999) / 10000000 * 10 else (nanos + 999999) / 1000000
     try Thread.sleep(sleepMs)
     catch {
-      case _: InterruptedException => Thread.currentThread.interrupt() // we got woken up
+      case _: InterruptedException => Thread.currentThread().interrupt() // we got woken up
```

Comment:
`currentThread` it doesn't change anything, it's essentially a magic getter so no parenthesises is fine IMO

Reply:
@johanandren @raboof  I don't think so.  Adding `( )` just denoting or tell other people that this method has side-effect or is not referential transparent. For instance, `Thread.currentThread` possibly generates two different results when invoking twice. Apart from `Thread#currentThread`, `System#nanoTime` or `System#currentTimeMills` , `Actor#sender()` also have side effects. In addtion, I noticed that in **akka-actor** module,  `ActorCell.scala`  **line 683**, `Thread.currentThread().interrupt()`

Reply:
In the early version of Akka, `sender` in `Actor` is sender, rather than `sender()`

Reply:
I think for us the interesting part to track is mutation since that may be unsafe in a multi-threaded context, so referential transparency is less interesting and so is side-effects that are just observing something non-pure. Since Akka is not a FP-pure code base I think we can be pragmatic about such things and do what is useful for us.

`interrupt()` is definitely something that should have parenthesises as it does mutate the interrupt flag of the thread.

Reply:
@patriknw What's your opinion about `Thread#currentThread`? Thanks :)

Reply:
I’d prefer currentThread() because it’s nor referential transparent, but no strong opinion.

Compare to ThreadLocalRandom.current() where it’s no doubt it should have () to discourage putting it in a val in constructor of for example an actor and then use the val in receive. currentThread is the same, although I can’t imagine anyone putting it in a val.

Reply:
Ok, that's a good point, I yield! ;) Let's go with `currentThread()`